### PR TITLE
Car 137 actions frontend components

### PR DESCRIPTION
**Pull request by alerayes**
This pull request contains most of the actions functionality. The calendar it's there, but it has no marked dates yet. 
I updated the ActionsScreen with the Stack.Navigator and I created three other screens: ActionsMain, ActionsType and ActionsCongrats.
The ActionsMain has the four images and see more buttons that leads to the the ActionType Screen. The ActionType Screen display all the actions and the user can select the ones he wants. Once the actions are selected, the submit button leads to the ActionsCongrats screen that display the added points and the total points.
The styling is not complete. 

Simulator Screen Shot - iPhone 14 Pro - 2023-03-15 at 16 39 41
Simulator Screen Shot - iPhone 14 Pro - 2023-03-15 at 16 39 32
Simulator Screen Shot - iPhone 14 Pro - 2023-03-15 at 16 39 23

### CODE REVIEW

File: screens/ActionsCongrats.tsx
Comment (luferupa) [inline screens/ActionsCongrats.tsx]: Why is this 'as any'?

File: screens/ActionsType.tsx
Comment (luferupa) [inline screens/ActionsType.tsx]: if you're using the ; , try to use it always  😅
It's just a recommendation

File: screens/ActionsType.tsx
Comment (luferupa) [inline screens/ActionsType.tsx]: You're not validating if we got a 200 response or an error. In case the backend generates an error, the app will break completely.

File: screens/ActionsType.tsx
Comment (luferupa) [inline screens/ActionsType.tsx:66]: Same problem, no errors caught. Or at least validate if the totalPoints come in the response.

File: screens/ActionsType.tsx
Comment (luferupa) [inline screens/ActionsType.tsx:9]: Why did you use this library component instead of the NativeBase one?

File: package.json
Comment (luferupa) [inline package.json]: Are we using this?

**Review by luferupa — COMMENTED** [review]
I added some comments in the files

File: screens/ActionsMain.tsx
Comment (KariGarcia) [inline screens/ActionsMain.tsx]: do we need these? or can we remove the comments?

File: screens/ActionsType.tsx
Comment (KariGarcia) [inline screens/ActionsType.tsx]: Lets remove these

File: screens/ActionsType.tsx
Comment (KariGarcia) [inline screens/ActionsType.tsx]: we are not doing anything with the response, is that intentional?

File: screens/ActionsType.tsx
Comment (KariGarcia) [inline screens/ActionsType.tsx]: lets remove console logs

File: screens/ActionsType.tsx
Comment (KariGarcia) [inline screens/ActionsType.tsx]: This is completely fine. But, you might want to consider not repeating as much as possible, here is a suggestion:
```suggestion
  const renderImage = () => {
    const images = {
       transport: {
         name: "test",
         alt: "Transport Image",
       },
       food: {
         name: "food",
         alt: "Food Image",
       },
       energies: {
         name: "energies",
         alt: "Energies Image",
       },
       products: {
         name: "products",
         alt: "Products Image",
       },
    };
    const image = images[actionType];
    if (!image) return null;
    return <Image 
          source={require(`../assets/images/${image.name}.png`)} 
          style={{justifyContent:'center'}}
          resizeMode='contain'
          alt={image.alt}
      />;
    }
```

**Review by KariGarcia — APPROVED** [review]
Looks good, just need to a bit of cleanup. Great job!

**Comment by KariGarcia** [comment]
Lets make sure we also get an ok from @luferupa before merging (:

File: screens/ActionsCongrats.tsx
Comment (alerayes) [reply screens/ActionsCongrats.tsx]: When I applied the styles without the any I got a typescript error.

File: screens/ActionsType.tsx
Comment (alerayes) [reply screens/ActionsType.tsx]: I tried to implement this one but I got an error.
iOS Bundling failed 84ms
error: screens/ActionsType.tsx: screens/ActionsType.tsx:Invalid call at line 108: require("../assets/images/" + image.name + ".png")

I checked the documentation and I found this:

Screen Shot 2023-03-16 at 4 45 54 PM

Can we leave it like this for now, and for the final I'll find as way to make it less repetitve.

File: package.json
Comment (alerayes) [reply package.json]: I forgot to uninstall it. I'll remove it from the dependecies.

File: screens/ActionsType.tsx
Comment (alerayes) [reply screens/ActionsType.tsx:9]: I used the expo checkbox because I was getting and error with the native base one.

File: screens/ActionsMain.tsx
Comment (alerayes) [reply screens/ActionsMain.tsx]: I'll remove it

File: screens/ActionsType.tsx
Comment (alerayes) [reply screens/ActionsType.tsx]: Yes. Should I remove the const resp and leave only the await server.post... ?

**Review by luferupa — APPROVED** [review]
Well done!

**Pull request merged**
